### DISCUSSION TITLE
How can the argType of type object be edited in the @addon-docs props table?

### DISCUSSION THREAD

**Discussion by dmarcanoalxn** [storybookjs/storybook#11649] [Help] [answered]
Given the following:

```
export default {
	title: "Example/Input",
	component: Input,
	argTypes: {
	  anObjectProperty: { control: { type: "object" }, defaultValue: { value: "my value" } },
	},
};

const Template = (args: any) => <Input {...args} />;

export const Custom: any = Template.bind({});
```

I would expect to be able to edit the object that I called anObjectProperty from the props table that the @addon-docs shows me. Nonetheless, when I try to edit the default object that I added, the field's border turns red, and it does not ever update the object.

Could anyone help me? ☹️

**Comment by shilman** [comment, marked as answer]
@dmarcanoalxn Yeah we need to redo the UX for this and documentation, but basically the `Object` control takes valid JSON as its input. So you should be able to edit something like:

```
{ "value": "my new value" }
```

**Reply by dmarcanoalxn** [reply]
Thanks for such a fast answer!

**Reply by MojoDeMedici** [reply]
This doesn't really work. It writes it as a string (Json): 
`"{\n  body: '#333333',\n  legs: '#232323)',\n  spots: '#454545'\n}"`
But the edit doesn't get submitted. I tried all day, and I cannot find a way to let this work.

In the ArgsType documentation I see a button (the first table in the doc) that says edit object, but I cannot figure out how implement it.

Note in my case the Object contains 3 colors, and is used to "paint" an svg (3 distinct paths).

I noticed that my object gets transformed in a string (the default value in the story is a string and not an object as it is in vue).

@shilman do you have idea of how let this work properly? I would love to be able to modify a color, and see it updated in real time on the screen (the docs are really hard to decipher, if you try a bit of customization).

Note (1) I am using Vue, and Vuetify, and my config is correct and everything works pretty smoothly (I am able to use also the css var generated by Vuetify).

Note (2) A couple of months ago I built a similar case, but using storybook 5 and knobs addon. I had it up and running in less than 5 mins.

**Reply by shilman** [reply]
@Dinuz can you file an issue with a repro?

**Reply by MojoDeMedici** [reply]
@shilman I think I solved the issue.
Apparently storybook is not able to use the default values directly, but needs to have them explicitly declared in the args. It's a bit weird, because if the props is a boolean, storybook sets the default value properly without any further declaration.

I understand that storybook is a react tool, but would be extremely valuable having examples and syntax description in the docs also for the other frameworks (that effectively are supported by storybook). I would help in writing them, but to be honest I don't have a very clear picture of how all works right now (I am keeping trying and guesting things). 

For example, today, I was trying to figure out, unsuccessfully, a way to get the canvas background of a particular story to be dark (the story is about a light components, that on white background is barely visible). I explored theming, but that is not what I want to do.
What I wanted to do was targeting the div with class sbdocs-preview, for the specific story, and change the color background of the class css-tmwzag. I could't find a way to accomplish that.

I guess the version 6 is way too new to find appropriate help in online communities, and I bet many other users are in similar conditions. This is why, tackling the documentation should be kind of imperative.

**Reply by MojoDeMedici** [reply]
@shilman never mind I didn't really solved the issue.
The same issue resurfaced in the moment in which I added a story as example.
This is what i get:
```
vue.js:634 [Vue warn]: Invalid prop: type check failed for prop "colors". Expected Object, got String with value "{
  body: 'var(--v-primary-base)',
  legs: 'var(--v-primary-darken2)',
  spots: 'var(--v-primary-darken3)'
}".
```

If I don't explicitly declare the value of the default props in args, I get the error.

To be precise, the error appears only when i switch tab from canvas to docs. Looks like the docs addon require an explicit declaration when working with objects (the objects gets transformed into json string).

**Reply by megagreg72** [reply]
Necro-ing this thread.  FYI, "object"-type control input works for me now, but only with double-quotes, such as 
```
{ "height" : 100 }
```
neither of these worked:
```
{'height' : 100 }
{height : 100 }
```

**Reply by shilman** [reply]
yes that's because it's supposed to be JSON. we've got a better UI coming here https://github.com/storybookjs/storybook/pull/12824

**Reply by igorcaldeira** [reply]
>  This doesn't really work. It writes it as a string (Json):
"{\n body: '#333333',\n legs: '#232323)',\n spots: '#454545'\n}"

Same problem here with controls, the default object value isn't valid
Hoping the better UI fix that

**Comment by AnnaCheba** [comment]
Hi, I have a similar problem, I want to be able to see values in that object with options to change it (similar to all other argTypes, with select boxes and boolean button ), at the moment it is just displaing json like object with option to change values as text, would love to have posibiity to set it as controls, like selectbox and so
  code example: 
```
argTypes: {
        optionsProp : {
            control:{
              type: 'object',
              productCode: {
                options: arrayOfCodes,
                control: {
                  type: 'select',
                },
               //....more argTypes under optionsProp object
                },
              },
            },
            table: {
                category: 'Formatting',
            },
        },
},
```

how it looks now:
<img width="536" alt="Screen Shot 2021-04-09 at 3 41 05 pm" src="https://user-images.githubusercontent.com/4457827/114134104-843aae80-994a-11eb-80cb-384d8ec9d3c8.png">
